src/render/src/factory.rs

```diff
@@ -234,6 +234,37 @@ pub trait FactoryExt<R: Resources>: Factory<R> {
         Ok(ShaderSet::Tessellated(vs, hs, ds, ps))
     }
 
+    /// Creates a `ShaderSet` from the supplied vertex, hull, domain, geometry and pixel
+    /// shader source code. Mainly used for testing.
+    fn create_shader_set_tessellation_with_geometry(&mut self, vs_code: &[u8], hs_code: &[u8], ds_code: &[u8], gs_code: &[u8], ps_code: &[u8])
+                         -> Result<ShaderSet<R>, ProgramError> {
+        let vs = match self.create_shader_vertex(vs_code) {
```

Comment:
let's be a tiny bit more modern and write this as `let vs = self.create_shader_vertex(vs_code).map_err(ProgramError::Vertex)?;`

Reply:
Done for all the `create_shader_*` convenience calls.